jerry-core/api/jerry.c

```diff
@@ -210,21 +210,31 @@ jerry_cleanup (void)
   }
 #endif /* JERRY_DEBUGGER */
 
-  for (jerry_context_data_header_t *this_p = JERRY_CONTEXT (context_data_p), *next_p = NULL;
+  for (jerry_context_data_header_t *this_p = JERRY_CONTEXT (context_data_p);
        this_p != NULL;
-       this_p = next_p)
+       this_p = this_p->next_p)
   {
-    next_p = this_p->next_p;
     if (this_p->manager_p->deinit_cb)
     {
       this_p->manager_p->deinit_cb (JERRY_CONTEXT_DATA_HEADER_USER_DATA (this_p));
     }
-    jmem_heap_free_block (this_p, sizeof (jerry_context_data_header_t) + this_p->manager_p->bytes_needed);
   }
 
   ecma_finalize ();
   jmem_finalize ();
   jerry_make_api_unavailable ();
```

Comment:
Oh shoot I need to move `jmem_finalize ()` to the end, because `jmem_heap_free_block` is used below.
Is that a problem?

Reply:
Don't think so.